tsdb/tx.go

```diff
@@ -316,7 +316,14 @@ func (l *LocalMapper) Begin(c *influxql.Call, startingTime int64, chunkSize int)
 	if fieldName != "" {
 		fid, err := l.decoder.FieldIDByName(fieldName)
 		if err != nil {
-			return fmt.Errorf("%s isn't a field on measurement %s", fieldName, l.job.MeasurementName)
+			switch {
+			case c.Name == "distinct":
```

Comment:
No saying we take them out, but when I see stuff special-cased, I just wondered why certain aggregates are handled nicely, and others aren't.

It's definitely no harm, and obviously adds to it. Not a blocker.